### ISSUE TITLE
Wrong preview for rotated screens

### ISSUE THREAD

**Issue by jlnr** [elementary/switchboard-plug-display#107]
I'm temporarily working with two screens, one of them physically rotated by 90° clockwise (=its contents need to be rotated by 270°).

The preview for the rotated screen is slightly wrong:

1. The screen rectangle is always wide, never tall, which makes it hard to align the screens vertically so that the mouse doesn't jump when moving from one screen to the other.
2. The label in the display preview is rotated 180° vs the label of my screen.
3. I have to choose "counter-clockwise" to get the correct rotation, however I would say my screen is rotated clock-wise. I'm not sure if that's really a bug, as it's all a matter of perspective, but it might cause #2.

![img_3134](https://user-images.githubusercontent.com/70772/43404180-2cc771be-9417-11e8-918e-37296a71c2d7.jpg)

**Comment by Djaler** [comment]
Are you on Loki. This is fixed already in Juno builds

**Comment by jlnr** [comment]
Fully patched Juno, although I haven‘t reinstalled in a while.

Sent from my iPhone

**Comment by Djaler** [comment]
Hmm, maybe this fix is only in daily repo

**Comment by jlnr** [comment]
https://github.com/elementary/switchboard-plug-display/issues/99#issuecomment-397824885

@Djaler Hmmm, are the screenshots in the last comment reversed? Shouldn't a display in "clockwise" rotation be higher than it is wide?

**Comment by jlnr** [comment]
Still happens for me on the current master branch.

**Comment by Djaler** [comment]
> Shouldn't a display in "clockwise" rotation be higher than it is wide?
I don't think so. We rotate an image on the screen. The rotation of the screen is not done by the system :smiley: 
And this is how it works now:
![screenshot from 2018-07-31 21 32 09](https://user-images.githubusercontent.com/7964583/43479643-86f20e22-9509-11e8-8e2d-052fd916b5f3.png)
![screenshot from 2018-07-31 21 34 48](https://user-images.githubusercontent.com/7964583/43479660-9049ced8-9509-11e8-82fa-714fde2cb5d9.png)

**Comment by jlnr** [comment]
I see. With a single screen, the current master branch is indeed consistent. The box doesn't rotate, but the label (representing the screen contents) does.

However, that approach can't handle setups with two screens or more. Spatially aligning the screens can only work when all screen contents have the same orientation in the preview, which is why the preview in my photo doesn't (and can't) match reality.

The way other systems handle this is to rotate the boxes, not the label. However I wouldn't mind if the label did in fact rotate, because then the monitor name in the preview would be aligned the same way as the "LG" branding on the right screen, etc.

**Comment by cassidyjames** [comment]
I definitely agree that the "rotation" should match the display, not the label. And I finally understand why this behavior seems backwards for about half of users no matter what we do! One school of thought is "rotating the image to match the display" which is probably more technically correct. The other is "rotate the display" which seems to be simpler and more how you would think if you were actually setting up multiple displays.

Because of that, I am also firmly in the "rotate the display" (which is the reverse of what the rotating the image would be... i.e. if I want to physically rotate a display 90 degrees clockwise, the system needs to rotate the image 90 degrees counter-clockwise). We should make sure we're always using that consistent mental model (and possible document it in the README or somewhere for future developer reference?), and figure out if we need to be more descriptive in the plug itself for users.
